cpp/include/stream_compaction.hpp

```diff
@@ -71,6 +82,28 @@ gdf_column apply_boolean_mask(gdf_column const &input,
  * @return gdf_column Column containing copy of all non-null elements of @p input.
  */
 gdf_column drop_nulls(gdf_column const &input);
+
+
+/**
+ * @brief Create a new table without duplicate rows 
+ *
+ * Given an input table, each row is copied to output table if the corresponding
+ * row of key column table is (unique or first duplicate row) or (unique or last
+ * duplicate row) or (only unique row).
+ *
+ * The input table and key columns table should have same number of rows.
```

Comment:
Sorry I didn't notice this earlier. Why is key columns not just an array of column indices? e.g. if you have 4 columns but you only want to consider uniqueness in the first two, you would pass [0, 1]. Why would you duplicate all rows of the columns?

Reply:
It's just pointers to the same columns, not actually duplicating the entire column.

Reply:
API is `table drop_duplicates(table input, table key_columns, ...`